config/initializers/1_rack_query_parser.rb

```diff
@@ -0,0 +1,8 @@
+require 'rack/request'
+require 'query_parser'
+
+Rack::Request::Env.class_eval do
+  def query_parser
```

Comment:
Is there an other way than monkey-patching?

Reply:
Sounds like a job for a Middleware?

Reply:
Yes, that was my idea last week on Slack, we even discussed this with @akofink, but the problem is that the `params` method is defined in the context of `ActionController` and not the context of a middleware, so you can't have access to it from a middleware. We could possibly alter the `request.query_string` in a middleware e.g. by replacing any occurrence `tags=` with `tags[]=`, but we would also need to keep in mind that `=` and `/` characters can be escaped in the tags and the params parser declared in Rails would just unescape everything equally.

This would mean that without introducing a new language (to bypass the implicit unescaping) we could end up with strings like `compliance=hey/david=yay/foo=bar=bar` where we would not know which one are the "real" `=` and `/` delimiters and which were the escaped one. Then in the controllers we need to parse this new language explicitly and also alter the logging to output the tags in a parsed forms instead of using the default (which again, would be this new language). I guess I don't have to explain why is this a bad practice :thinking: